StormDBManager.py

```diff
@@ -0,0 +1,254 @@
+import logging
+from storm.database import Connection, create_database
+from storm.exceptions import OperationalError
+
+from storm.twisted.transact import Transactor, transact
+from twisted.internet.defer import DeferredLock, inlineCallbacks
+
+from twisted.internet import reactor
+
+
+class StormDBManager:
+    """
+    The StormDBManager is a manager that runs queries using the Storm Framework.
+    These queries will be run on the Twisted thread-pool to ensure asynchronous, non-blocking behavior.
+    In the future, this database manager will be the basis to an ORM based approach.
+    """
+
+    def __init__(self, db_path):
+        """
+        Sets up the database and all necessary elements for the
+        database manager to function.
+        """
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        self.db_path = db_path
+
+        # The transactor is required when you have methods decorated with the @transact decorator
+        # This field name must NOT be changed.
+        self.transactor = Transactor(reactor.getThreadPool())
+
+        # Create a DeferredLock that should be used by callers to schedule their call.
+        self.db_lock = DeferredLock()
+
+    @inlineCallbacks
+    def initialize(self):
+        """
+        Opens/creates the database and initializes the version.
+        """
+        # Open or create the database
+        self._database = create_database(self.db_path)
+        self.connection = Connection(self._database)
+        self._version = 0
+        yield self._retrieve_version()
+
+    @property
+    def version(self):
+        return self._version
+
+    @inlineCallbacks
+    def _retrieve_version(self):
+        """
+        Attempts to retrieve the current datbase version from the MyInfo table.
+        If it fails, the _version field remains at 1 as defined in the init function.
```

Comment:
Are you referencing to the `initialize` method here? This initialises `_version` to 0.